src/coreComponents/unitTests/wellsTests/testOpenClosePerf.cpp

```diff
@@ -0,0 +1,573 @@
+                                                                             string_array const & regionNames )
+    {
+      mesh.getElemManager().forElementSubRegions< WellElementSubRegion >( regionNames,
+                                                                          [&]( localIndex const,
+                                                                               WellElementSubRegion & subRegion )
+      {
+        array1d< integer > & currentStatus = subRegion.getWellElementStatus();
+        EXPECT_EQ( refStatus.size(), currentStatus.size() );
+        const integer * curStat = currentStatus.data();
+        for( integer i = 0; i < currentStatus.size(); ++i )
+        {
+          EXPECT_EQ( curStat[ i ], refStatus[ i ] );
+        }
+      } );
+    } );
+    it++;
+  }
+}
+
+template< typename LAMBDA >
+void testPlugBottomUpPerfCheck( CompositionalMultiphaseReservoirAndWells<> & solver,
+                                DomainPartition & domain,
+
+                                LAMBDA && perfFunction )
+{
+  CompositionalMultiphaseWell & wellSolver = *solver.wellSolver();
+
+  typedef std::map< real64, std::vector< int > > map_type;
+  map_type refVal;
+  refVal[4800.0] =  { 1, 1, 1, 1, 1};
+  refVal[14800.0] = { 1, 1, 1, 1, 0};
+  refVal[19600.0] = { 1, 1, 1, 0, 0};
+  refVal[32400.0] = { 1, 1, 0, 0, 0};
+  refVal[39800.0] = { 0, 0, 0, 0, 0};
+
+  map_type::iterator it = refVal.begin();
+  while( it != refVal.end() )
+  {
+    perfFunction( it->first );
+
+    const std::vector< int > & refStatus = it->second;
+    wellSolver.forDiscretizationOnMeshTargets( domain.getMeshBodies(), [&] ( string const &,
+                                                                             MeshLevel & mesh,
+                                                                             string_array const & regionNames )
+    {
+      mesh.getElemManager().forElementSubRegions< WellElementSubRegion >( regionNames,
+                                                                          [&]( localIndex const,
+                                                                               WellElementSubRegion & subRegion )
+      {
+        array1d< integer > & currentStatus = subRegion.getWellElementStatus();
+        EXPECT_EQ( refStatus.size(), currentStatus.size() );
+        const integer * curStat = currentStatus.data();
+        for( integer i = 0; i < currentStatus.size(); ++i )
+        {
+          EXPECT_EQ( curStat[ i ], refStatus[ i ] );
+        }
+      } );
+    } );
+    it++;
+  }
+}
+
+template< typename LAMBDA >
+void testOpenTopDownPerfCheck( CompositionalMultiphaseReservoirAndWells<> & solver,
+                               DomainPartition & domain,
+
+                               LAMBDA && perfFunction )
+{
+  CompositionalMultiphaseWell & wellSolver = *solver.wellSolver();
+
+  typedef std::map< real64, std::vector< int > > map_type;
+  map_type refVal;
```

Comment:
To make it clearer, what do you think of:
```suggestion
  map_type refPerfTable;
```